### PR TITLE
Centralize the docindex scripts and regenerate Github IO landing page.

### PR DESCRIPTION
**Pull request by sima-zhu**

### CODE REVIEW

File: eng/pipelines/create-doc-index.yml [hunk 1]
```diff
@@ -15,25 +15,24 @@ jobs:
...
-          sphinx-build -b html -c $(Build.SourcesDirectory)/doc/sphinx $(Build.SourcesDirectory)/_docs $(Build.SourcesDirectory)/_docs/html
-          Copy-Item -Path $(Build.SourcesDirectory)/_docs/html/* -Destination $(Build.ArtifactStagingDirectory) -Recurse -Force
-        displayName: 'Generate Documentation Index'
-
+          Invoke-WebRequest -MaximumRetryCount 10 -Uri "https://github.com/dotnet/docfx/releases/download/v2.43.2/docfx.zip" `
+          -OutFile "docfx.zip" | Wait-Process; Expand-Archive -Path "docfx.zip" -DestinationPath "./docfx/"
+          echo "##vso[task.setvariable variable=docfxPath;isOutput=true]$(Build.SourcesDirectory)/docfx/docfx.exe"
+        workingDirectory: $(Build.SourcesDirectory)
+        displayName: Download and Extract DocFX
+        name: setupDocfxTool
```
Comment (weshaggard) [inline eng/pipelines/create-doc-index.yml:24]: What is this name for?

File: eng/docgeneration/Generate-DocIndex.ps1 [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    Copy-Item "$($RepoRoot)/CONTRIBUTING.md" -Destination "${DocOutDir}/api/CONTRIBUTING.md" -Force
+
+    LogDebug "Building site..."
+    & $($DocFx) build "${DocOutDir}/docfx.json"
+    # The line below is used for testing in local
+    #docfx build "${DocOutDir}/docfx.json"
+    Copy-Item "${DocGenDir}/assets/logo.svg" -Destination "${DocOutDir}/_site/" -Force    
+}
+
+if ((Get-ChildItem -Path Function: | ? { $_.Name -eq $GetGithubIoDocIndexFn  }).Count -gt 0)
```
Comment (weshaggard) [inline eng/docgeneration/Generate-DocIndex.ps1:143]: Test-Path?

File: eng/pipelines/create-doc-index.yml (hunk 1 above)
Comment (sima-zhu) [reply eng/pipelines/create-doc-index.yml:24]: Has been discussed in another thread

**Review by weshaggard — APPROVED** [review]
The change looks good. Lets please make sure the generation looks correct and do chat with Scott to figure out what things we can clean-up now.

File: eng/pipelines/create-doc-index.yml [hunk 3]
```diff
@@ -15,26 +15,65 @@ jobs:
       vmImage: 'vs2017-win2016'
 
     steps:
+      - pwsh: |
+          Invoke-WebRequest -MaximumRetryCount 10 -Uri "https://github.com/dotnet/docfx/releases/download/v2.43.2/docfx.zip" `
+          -OutFile "docfx.zip" | Wait-Process; Expand-Archive -Path "docfx.zip" -DestinationPath "./docfx/"
+          echo "##vso[task.setvariable variable=docfxPath;isOutput=true]$(Build.BinariesDirectory)/docfx/docfx.exe"
```
Comment (weshaggard) [inline eng/pipelines/create-doc-index.yml:21]:
```suggestion
          echo "##vso[task.setvariable variable=docfxPath]$(Build.BinariesDirectory)/docfx/docfx.exe"
```

File: eng/scripts/Language-Settings.ps1 [hunk 4]
```diff
@@ -104,4 +105,19 @@ function Publish-python-GithubIODocs ($DocLocation, $PublicArtifactLocation)
     $releaseTag = RetrieveReleaseTag "PyPI" $PublicArtifactLocation
     Upload-Blobs -DocDir $UnzippedDocumentationPath -PkgName $PkgName -DocVersion $Version -ReleaseTag $releaseTag
   }
+}
+
+function Get-python-GithubIoDocIndex() {
+  # Fetch out all package metadata from csv file.
+  $metadata = Get-CSVMetadata -MetadataUri $MetadataUri
+  # Leave the track 2 packages if multiple packages fetched out.
+  # $clientPackages = $metadata | Where-Object { $_.GroupId -eq 'com.azure' }
```
Comment (weshaggard) [inline eng/scripts/Language-Settings.ps1:114]: I assume these should be removed.

**Review by weshaggard — APPROVED** [review]

**Pull request merged**